sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAPITests.groovy

```diff
@@ -1052,6 +1053,274 @@ class FileAPITests extends APISpec {
         assertExceptionStatusCodeAndMessage(e, 400, ShareErrorCode.INVALID_HEADER_VALUE)
     }
 
+    @Unroll
+    def "Start copy with options"() {
+        given:
+        primaryFileClient.create(1024)
+        def sourceURL = primaryFileClient.getFileUrl()
+        def filePermissionKey = shareClient.createPermission(filePermission)
+        // We recreate file properties for each test since we need to store the times for the test with namer.getUtcNow()
+        smbProperties.setFileCreationTime(namer.getUtcNow())
+            .setFileLastWriteTime(namer.getUtcNow())
+        if (setFilePermissionKey) {
+            smbProperties.setFilePermissionKey(filePermissionKey)
+        }
+        def options = new ShareFileCopyOptions()
+        .setSmbProperties(smbProperties)
+            .setFilePermission(setFilePermission ? filePermission : null)
+            .setIgnoreReadOnly(ignoreReadOnly)
+            .setSetArchiveAttribute(setArchiveAttribute)
+            .setPermissionCopyModeType(permissionType)
+
+        when:
+        SyncPoller<ShareFileCopyInfo, Void> poller = primaryFileClient.beginCopy(sourceURL, options)
+
+        def pollResponse = poller.poll()
+
+        then:
+        pollResponse.getValue().getCopyId() != null
+        pollResponse.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED
+
+        where:
+        setFilePermissionKey | setFilePermission | ignoreReadOnly | setArchiveAttribute | permissionType
+        true                 | false             | false          | false               | PermissionCopyModeType.OVERRIDE
+        false                | true              | false          | false               | PermissionCopyModeType.OVERRIDE
+        false                | false             | true           | false               | PermissionCopyModeType.SOURCE
+        false                | false             | false          | true                | PermissionCopyModeType.SOURCE
+    }
+
+    def "Start copy with options IgnoreReadOnly and SetArchive"() {
+        given:
+        primaryFileClient.create(1024)
+        def sourceURL = primaryFileClient.getFileUrl()
+        def options = new ShareFileCopyOptions()
+            .setIgnoreReadOnly(true)
+            .setSetArchiveAttribute(true)
+
+        when:
+        SyncPoller<ShareFileCopyInfo, Void> poller = primaryFileClient.beginCopy(sourceURL, options)
+
+        def pollResponse = poller.poll()
+
+        then:
+        pollResponse.getValue().getCopyId() != null
+        pollResponse.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED
+    }
+
+    def "Start copy with options file permission"() {
```

Comment:
Are most of these tests duplicates of existing tests but using the options?

Reply:
Yup basically, I'm just asserting the results of passing the necessary fields in options. Is this redundant?

Reply:
Personally I think it is, yes, and would only add tests for the new feature being added